src/main/java/com/starrocks/connector/spark/cfg/ConfigurationOptions.java

```diff
@@ -67,4 +79,44 @@ public interface ConfigurationOptions {
 
     String STARROCKS_DESERIALIZE_QUEUE_SIZE = "starrocks.deserialize.queue.size";
     int STARROCKS_DESERIALIZE_QUEUE_SIZE_DEFAULT = 64;
+
+    static Map<String, String> makeWriteCompatibleWithRead(Map<String, String> options) {
+        // user and password compatible
+        Map<String, String> configMap = new HashMap(options);
+        processSameCompatible(ImmutableSet.of(STARROCKS_USER, STARROCKS_REQUEST_AUTH_USER, "user"), configMap);
+        processSameCompatible(ImmutableSet.of(STARROCKS_PASSWORD, STARROCKS_REQUEST_AUTH_PASSWORD, "password"), configMap);
+        processSameCompatible(ImmutableSet.of(STARROCKS_FENODES, KEY_FE_HTTP), configMap);
+        return configMap;
+    }
+
+    static boolean processSameCompatible(Set<String> confCandidates, Map<String, String> options) {
+        String value = "";
+        for (String key : confCandidates) {
+            if (options.containsKey(key)) {
+                value = options.get(key);
+            }
+        }
+        if (StringUtils.isNotEmpty(value)) {
```

Comment:
why not use not null as the condition. empty may be valid, for example, password can be empty